client/pages/Login/Invitation/Invitation.tsx

```diff
@@ -9,46 +9,68 @@ import { useTranslation } from 'react-i18next'
 
 import { Objects } from '@core/utils'
 import { BasePaths } from '@client/basePaths'
+import { ApiEndPoint } from '@common/api/endpoint'
 
 const Invitation: React.FC = () => {
   const dispatch = useAppDispatch()
+  const { i18n } = useTranslation()
   const history = useHistory()
-  const invitedUser = useAppSelector((state) => state.login.invitedUser)
-  const invitation = Urls.getRequestParam('invitation')
   const loggedUser = useUser()
 
-  const { i18n } = useTranslation()
+  const invitationUuid = Urls.getRequestParam('invitationUuid')
+  const { userRole, assessment, user: invitedUser } = useAppSelector((state) => state.login.invitation)
 
   useEffect(() => {
-    if (invitation) {
-      dispatch(LoginActions.fetchUserByInvitation(invitation))
+    if (invitationUuid) {
+      dispatch(LoginActions.fetchUserByInvitation(invitationUuid))
     }
   }, [])
 
   const onAccept = () => {
-    dispatch(LoginActions.acceptInvitation(invitation))
+    dispatch(LoginActions.acceptInvitation(invitationUuid))
     history.push(BasePaths.Root())
   }
 
-  if (Objects.isEmpty(invitation)) {
+  if (Objects.isEmpty(invitationUuid) || !invitedUser) {
```

Comment:
if that's the case, we need a redirect to 404

Reply:
it redirects to the homepage - history.push(BasePaths.Root())